src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,45 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.utils.Storage;
+import duke.utils.TaskList;
+import duke.utils.UI;
+
+/**
+ * A Command task used by the app.
+ */
+public abstract class Command {
+    protected Storage storage;
+    protected UI ui;
+    protected TaskList taskList;
+
+    /**
+     * Constructs a specified Command command for Duke.
+     *
+     * @param storage Storage of the current Duke program.
+     * @param ui UI of the current Duke program.
+     * @param taskList Tasklist of the current Duke program.
+     */
+    public Command(Storage storage, UI ui, TaskList taskList) {
+        this.storage = storage;
+        this.ui = ui;
+        this.taskList = taskList;
+    }
+
+    /**
+     *  Constructs a specified Command task for Duke that does not require any Duke utility classes.
+     */
+    public Command() {
```

Comment:
Perhaps this constructor is not necessary since Java will automatically make one for you if the desired constructor is as empty constructor.